File: src/main/kotlin/com/full/stack/assignment/f1/model/Circuit.kt
package com.full.stack.assignment.f1.model

import com.fasterxml.jackson.annotation.JsonProperty

data class Circuit(
    @JsonProperty("circuitId")
    val id: String,
    @JsonProperty("circuitName")
    val name: String,
    @JsonProperty("Location")
    val location: CircuitLocation
)

data class CircuitLocation(
    @JsonProperty("locality")
    val locality: String,
    @JsonProperty("country")
    val country: String
)

File: frontend/lib/domain/models/circuit/circuit.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'circuit.freezed.dart';

@freezed
abstract class Circuit with _$Circuit {
  const factory Circuit({
    required String circuitId,
    required String circuitName,
    required String locality,
    required String country,
  }) = _Circuit;
}
